flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactoryTest.java

```diff
@@ -61,15 +61,23 @@
 
 	@Test
 	public void testInterruptsNotCached() throws Exception {
-		ConnectionID connectionId = new ConnectionID(new InetSocketAddress(InetAddress.getLocalHost(), 8080), 0);
-		try (AwaitingNettyClient nettyClient = new AwaitingNettyClient()) {
+		final NettyTestUtil.NettyServerAndClient nettyServerAndClient = createNettyServerAndClient();
+		ConnectionID connectionId = new ConnectionID(new InetSocketAddress(
+			nettyServerAndClient.server().getConfig().getServerAddress(),
+			nettyServerAndClient.server().getConfig().getServerPort()),
+			0);
+		try {
+			AwaitingNettyClient nettyClient = new AwaitingNettyClient(nettyServerAndClient.client());
 			PartitionRequestClientFactory factory = new PartitionRequestClientFactory(nettyClient, 0);
 
 			nettyClient.awaitForInterrupts = true;
 			connectAndInterrupt(factory, connectionId);
 
 			nettyClient.awaitForInterrupts = false;
 			factory.createPartitionRequestClient(connectionId);
+		} finally {
+			nettyServerAndClient.client().shutdown();
+			nettyServerAndClient.server().shutdown();
 		}
```

Comment:
Couldn't you leave it as is and just request a `ConnectionID` from `AwaitingNettyClient`?

Reply:
I did it at first but it didn't look to me like a right abstraction (a `client` wrapping `serverAndClient` and exposing a `ConnectionID`).

Reply:
Okay fair point. How about, keep as is, add some `AwaitingNettyClient#getSocketAddress` and use that to setup connectionId inside the try-with-resource block?

Reply:
The part "`client` wrapping `serverAndClient`"
would still hold (and I'd like to avoid it).

Why do you want to stick with the current way?

Reply:
I like the try-with-resource approach more - it makes the code cleaner and test easier to understand. But you'd have to move the `AutoCloseable` to `NettyServerAndClient`, which is probably out of scope here. I also now fully understand your argument.